### PR TITLE
Refs #16438 - extract code to foreman-remote-execution

### PR DESCRIPTION
**Pull request by iNecas**
The dependency from smart_proxy_remote_execution_ssh changes to
foreman_remote_execution_core. The behaviour should stay the same.

### CODE REVIEW

**Comment by iNecas** [comment]
Requires https://github.com/theforeman/foreman_remote_execution/pull/203

File: README.md [hunk 1]
```diff
@@ -13,8 +13,10 @@ This a plugin for foreman smart-proxy allowing using ssh for the
 Add this line to your smart proxy bundler.d/ssh.rb gemfile:
 
 ```ruby
-gem 'smart_proxy_dynflow', :git => 'https://github.com/iNecas/smart_proxy_dynflow.git'
-gem 'smart_proxy_ssh', :git => 'https://github.com/iNecas/smart_proxy_ssh.git'
+gem 'smart_proxy_dynflow', :git => 'https://github.com/theforeman/smart_proxy_dynflow.git'
+gem 'smart_proxy_dynflow_core', :git => 'https://github.com/theforeman/smart_proxy_dynflow.git'
+gem 'foreman_remote_execution_core', :git => 'https://github.com/theforeman/foreman_remote_execution.git'
+gem 'smart_proxy_ssh', :git => 'https://github.com/theforeman/smart_proxy_ssh.git'
```
Comment (adamruzicka) [inline README.md]: `s/_s/_remote_execution_s/g`

Comment (iNecas) [inline README.md]: nice catch!

**Comment by iNecas** [comment]
Merged as https://github.com/theforeman/smart_proxy_remote_execution_ssh/commit/cb87b94e058dcd95892d38306d6cafe12ce2a860

**Pull request closed without merging**